core/src/model.jl

```diff
@@ -172,6 +172,10 @@ function Model(config::Config)::Model
     @debug "Read database into memory."
 
     u0 = build_state_vector(parameters.p_non_diff)
+    if length(u0) == 0
+        error("Models without states are unsupported, please add a Basin node.")
+    end
```

Comment:
```suggestion
    if isempty(u0)
        @error "Models without states are unsupported, please add a Basin node."
        error("Model has no state.")
    end
```

This is a bit of a weird pattern but we tend to use this. The error macro throws a clear user facing error message that is not fatal, execution continues after. During validation we tend to log all the errors first, and the throw an error saying validation failed.

In this case we want to stop immediately so an error log is followed by an error.

Perhaps in a separate PR we can look into improving this, with e.g. a fatal log that throws, or deciding that the error message on top of the stacktrace is readable enough in the log file and stdout that we can only error.